Scripts/LinearRegression.R

```diff
@@ -35,42 +35,45 @@ main <- function(path, datafilename){
   
   
   ##### Aggregate Daily Average
-  airq_daily = airq %>%
+  airq_daily = data %>%
     group_by(Date) %>%
     summarise_all(funs(mean), na.rm = TRUE)
   
+  ##### Regressions for ALL chemicals
+  regressions = airq_daily %>% 
+    select(-c(Date, Time, Date_time, Temp, RH, AH )) %>%  # leave only dependent variables
+    map(~lm(.x ~  Temp + AH, data = airq_daily)) %>% 
+    map(tidy) 
   
-  ##### REGRESSIONs
+  ##### Export your model object to an RDS file
   
-  #Fit benzene against temperature & humidity (relative humidity is correlated to temperature) 
-  Benzene = lm(`C6H6(GT)` ~ T + AH, data = airq_daily)
-  #save RDS file
-  saveRDS(Benzene, file = "Benzene.rds")
-  tidy(Benzene)
+  chemnames <- c("CO", "Tin_oxide", "Hydro_carbons", "Benzene", 
+                "Titania", "NOx", "Tungsten_oxide_NOx", "NO2", "Tungsten_oxide_NO2", 
+                "Indium_oxide")
   
-  #Fit titania against temperature & humidity 
-  Titania = lm(`PT08.S2(NMHC)` ~ T + AH, data = airq_daily)
-  #save RDS file
-  saveRDS(Titania, file = "Titania.rds")
-  tidy(Titania)
-  
-  #Fit tin oxide against temperature & humidity 
-  Tinoxide= lm(`PT08.S1(CO)` ~ T + AH, data = airq_daily)
-  #save RDS file
-  saveRDS(Tinoxide, file = "Tinoxide.rds")
-  tidy(Tinoxide)
+  for (i in 1:length(regressions)) {
+    assign(chemnames[[i]], regressions[[i]])
+    saveRDS(regressions[[i]], file = paste(chemnames[[i]]))
```

Comment:
`file = here('Data', paste(chemnames[[i]])))` ?

They should be saved to the data directory.